packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -0,0 +1,99 @@
+import { assetDataUtils, SignedOrder } from '@0x/order-utils';
+import { ERC20AssetData } from '@0x/types';
+import { BigNumber, logUtils } from '@0x/utils';
+import Axios, { AxiosResponse } from 'axios';
+
+import { MarketOperation } from '../types';
+
+/**
+ * Request quotes from RFQ-T providers
+ */
+export class QuoteRequestor {
+    private readonly _rfqtMakerEndpoints: string[];
+
+    constructor(rfqtMakerEndpoints: string[]) {
+        this._rfqtMakerEndpoints = rfqtMakerEndpoints;
+    }
+
+    public async requestRfqtFirmQuotesAsync(
+        makerAssetData: string,
+        takerAssetData: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        intentOnFilling: boolean,
+        takerApiKey: string,
+        takerAddress: string,
+    ): Promise<SignedOrder[]> {
+        const makerEndpointMaxResponseTimeMs = 1000;
+
+        const getTokenAddressOrThrow = (assetData: string): string => {
```

Comment:
Why is this a closure? It could be useful elsewhere.

Reply:
I just wrote it "as local as possible."  Figured we'd graduate it if/when it becomes useful elsewhere.  (Though how we'd remember it exists when it comes time to use it is undefined.)  I'd be happy to move it wherever you see fit.

Reply:
I think it makes sense to just make this a regular, unexported function in this file.

Reply:
Addressed in https://github.com/0xProject/0x-monorepo/pull/2541/commits/39c2a757be66359f602400b610a8f303faf01f2f